texastribune/donations-app
# REVAMP: Ready for deployment

#### What's the plan?
I think you know.

#### What's this PR do?
Minus our fun little dots, which I'll code over the weekend, the JavaScript portion of this is finished. That leaves styling for next sprint.

I also updated the README on this branch to explain the new dev and build processes.

You can skip to the manual testing steps now.

#### Why are we doing this? How does it help us?

#### How should this be manually tested?
1. Get up and running by following the steps in the new README. If you encounter issues, let me know, because that probably means my documentation sucks.
2. `npm run dev'
3. In your browser, you should see a working -- yet incredibly hideous -- carousel form. Play around with changing the frequencies, ranges and amounts. Is everything dynamically updating as outlined in Ben's mockups? When you click the submit button, is it taking you to the checkout page with the proper value and frequency?
4. Check out my JS. `raw/js/form.js` is where all the logic lives. `raw/js/index.js` is the Webpack entry point that initializes the `FormHandler` class. Does the logic make sense? Is it modular enough?
5. Lastly, if you so desire, check out the tests in `tests/`. You can run them with `npm test`. Do they make sense? Anything I should be checking that I'm not?

 The form class is initialized in `raw/js/index.js`.

#### Have automated tests been added?

#### Are there performance implications? If adding JS can it be done async? Do images/CSS/JS have cache headers set?

#### What are the relevant tickets?

#### How should this change be communicated to end users?

#### Next steps?

#### Smells?

#### TODOs:

#### Has the relevant documentation/wiki been updated?

#### Technical debt note
